contracts/Nexus.sol

```diff
@@ -193,6 +199,29 @@ contract Nexus is INexus, BaseAccount, ExecutionHelper, ModuleManager, UUPSUpgra
         }
     }
 
+    function emergencyUninstallHook(address hook, bytes calldata deInitData) external payable onlyEntryPoint {
+        AccountStorage storage accountStorage = _getAccountStorage();
+        uint256 hookTimelock = accountStorage.emergencyUninstallTimelock[hook];
+
+        if (hookTimelock == 0) {
+            // if the timelock hasnt been initiated, initiate it
+            accountStorage.emergencyUninstallTimelock[hook] = block.timestamp;
+            emit EmergencyHookUninstallRequest(hook, block.timestamp);
+        } else if (block.timestamp >= hookTimelock + 3 * _EMERGENCY_TIMELOCK) {
+            // if the timelock has been left for too long, reset it
```

Comment:
what is the reason for this?
can we not just allow uninstall here?

Reply:
yeah I was thinking through this and we could allow these timelocks to not expire. The main upside of this is that when you initiate a timelock and the waiting time (1 day) has elapsed, the account is in a state where the hook can always be instantly uninstalled, which kind of defeats the purpose of the timelock as a guarantee that uninstallation cannot be instant